src/main/java/core/basesyntax/BalanceOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
```

Comment:
Please ensure that the `handle` method is following the encapsulation principle. The `storage` map should not be directly exposed to manipulation. Consider providing a more controlled way of interacting with the storage, such as through a dedicated service or by encapsulating the storage logic within this class.